### PR TITLE
Add TQFN-48-1EP_7x7mm_P0.5mm_EP5.1x5.1mm

### PR DESCRIPTION
**Pull request by jneiva08**
Add footprint for TQFN-48 (T4877-3)

footprint Datasheet: https://pdfserv.maximintegrated.com/package_dwgs/21-0144.PDF
IC Datasheet: https://datasheets.maximintegrated.com/en/ds/MAX14830.pdf

### CODE REVIEW

**Comment by chschlue** [comment]
Pending #422

**Comment by jneiva08** [comment]
@chschlue how is the best way to resolve this conflict?

**Comment by evanshultz** [comment]
@jneiva08 
Try to merge the master branch on your fork (make sure it is up to date first) into this branch. if that doesn't work, you'll have to manually update the YAML file.

**Comment by jneiva08** [comment]
@evanshultz have resolved the conflict, hope i didn't miss anything

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml [hunk 1]
```diff
@@ -710,3 +710,62 @@ TQFN-40-1EP_5x5mm_P0.40mm_EP3.50x3.50mm:
...
+TQFN-48-1EP_7x7mm_P0.5mm_EP5.1x5.1mm:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0144.PDF' #T4877-3
+  ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  # custom_name_format:
+  body_size_x:
+    nominal: 7
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml]: Give the min and max as well. Below also.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml [hunk 2]
```diff
@@ -710,3 +710,62 @@ TQFN-40-1EP_5x5mm_P0.40mm_EP3.50x3.50mm:
...
+  body_size_y:
+    nominal: 7
+  overall_height:
+    minimum: 0.7
+    nominal: 0.75
+    maximum: 0.8
+
+  lead_width:
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml]: This is wrong. Please check again.

**Comment by evanshultz** [comment]
@jneiva08 
Thanks!

I added a few comments above, and the paste coverage is rather high. `0.65` to `0.7` should be alright. But this percentage is used on most footprints in this YAML file. Probably will still be OK. Up to you.

**Comment by jneiva08** [comment]
@evanshultz have made the modifications you have mentioned.
The paste coverage used is the same as others footprints? should be altered to [4x4] or [2x2] ?
Sorry I don have any clue about this.

**Comment by chschlue** [comment]
Paste coverage is the percentage of pad area that will be covered with solder paste during assembly, i.e. the stencil cutout in relation to the pad's copper.

True, most size definitions use 0.75. I guess in most cases this crept in by copying and pasting a definition using it intentionally.

**Comment by evanshultz** [comment]
@jneiva08 
Thanks! We can go with this, in light of paste coverage having some leeway and not being captured in the footprint name.

Please re-generate the footprint and update the PR for it, and then I can merge both PRs.

**Comment by jneiva08** [comment]
@evanshultz only could re-generate the footprint today.

**Comment by jneiva08** [comment]
thanks for your time

**Pull request merged**